diff --git a/Android.mk b/Android.mk
index 0ffb7fe..4c44c41 100644
--- a/Android.mk
+++ b/Android.mk
@@ -66,7 +66,11 @@ include $(multirom_local_path)/trampoline/Android.mk
 include $(multirom_local_path)/install_zip/Android.mk
 
 # Kexec-tools
-include $(multirom_local_path)/kexec-tools/Android.mk
+ifeq ($(TARGET_ARCH),arm64)
+    include $(multirom_local_path)/kexec-tools-arm64/Android.mk
+else
+    include $(multirom_local_path)/kexec-tools/Android.mk
+endif
 
 # adbd
 include $(multirom_local_path)/adbd/Android.mk
